Content.Server/Store/Systems/StoreSystem.cs

```diff
@@ -69,10 +69,12 @@ private void OnStoreOpenAttempt(EntityUid uid, StoreComponent component, Activat
         if (!component.OwnerOnly)
             return;
 
-        component.AccountOwner ??= args.User;
+        _mind.TryGetMind(args.User, out var mind, out var mindComp);
+
+        component.AccountOwner ??= mind;
         DebugTools.Assert(component.AccountOwner != null);
 
-        if (component.AccountOwner == args.User)
+        if (component.AccountOwner == mind)
             return;
 
         _popup.PopupEntity(Loc.GetString("store-not-account-owner", ("store", uid)), uid, args.User);
```

Comment:
We should fall back to the body if mind is null.
Also for fun just double check to make sure this works with all purchases, Revenant, and traitors getting their uplinks assigned.